examples/platform/telink/build_info.cmake

```diff
@@ -0,0 +1,148 @@
+#
+#    Copyright (c) 2024 Project CHIP Authors
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+# Matter git info
+execute_process(
+    COMMAND git rev-parse HEAD
```

Comment:
what is the timing of these commands? 

We have added some `place SHA of the repo during build` logic and the end result was that the build was slowed down significantly. I would rather we did not run git commands during build time.

Reply:
Maybe make it a non-default build option. If I build things for my local validation of builds, I do not want to spend extra time for the build.

Reply:
@andy31415 all it took for commands from `build_info.cmake` is 1sec

> -- Version info get start time: 1730879694
-- End Time: 1730879695
-- Execution Time: 1 seconds

I agree with your point to put all of this under some condition/option, will discuss with team's needs.

Reply:
Made present only for `CONFIG_CHIP_APP_LOG_LEVEL > 3` (debug and verbose)